src/transformers/modeling_longformer.py

```diff
@@ -113,137 +114,167 @@ def __init__(self, config, layer_id):
             attention_window > 0
         ), f"`attention_window` for layer {self.layer_id} has to be positive. Given {attention_window}"
 
-        self.one_sided_attention_window_size = attention_window // 2
+        self.one_sided_attn_window_size = attention_window // 2
 
     @staticmethod
-    def _skew(x, direction):
-        """Convert diagonals into columns (or columns into diagonals depending on `direction`"""
-        x_padded = F.pad(x, direction)  # padding value is not important because it will be overwritten
-        x_padded = x_padded.view(*x_padded.size()[:-2], x_padded.size(-1), x_padded.size(-2))
-        return x_padded
+    def pad_and_transpose_last_two_dims(hidden_states_padded, padding):
+        """Convert diagonals into columns or columns into diagonals depending on `padding`"""
+        hidden_states_padded = F.pad(
+            hidden_states_padded, padding
+        )  # padding value is not important because it will be overwritten
+        hidden_states_padded = hidden_states_padded.view(
+            *hidden_states_padded.size()[:-2], hidden_states_padded.size(-1), hidden_states_padded.size(-2)
+        )
+        return hidden_states_padded
 
     @staticmethod
-    def _skew2(x):
+    def _pad_by_window_overlap_except_last_row(chunked_hidden_states):
         """shift every row 1 step to right converting columns into diagonals"""
```

Comment:
```suggestion
        """shift every row 1 step right,  converting columns into diagonals"""
```

I don't understand what it means to convert a column to a diagonal.

Reply:
should the func name be shift_hid_states_right?